### PR TITLE
docs: Revamp how-to guides structure

### PR DESCRIPTION
**Pull request by afgambin**
Addressing Jira issue: https://warthogs.atlassian.net/browse/KF-5814

### CODE REVIEW

**Comment by afgambin** [comment]
@kimwnasptd this is ready for review. Full details of the PR can be found in the Jira issue attached in the description. Note that "List DSS commands" section is not modified as I expect to move that information to the [API Reference guide we plan to work on](https://warthogs.atlassian.net/browse/KF-5816).

**Comment by afgambin** [comment]
> Note that "List DSS commands" section is not modified as I expect to move that information to the [API Reference guide we plan to work on](https://warthogs.atlassian.net/browse/KF-5816).

Actually, after having another look, the `List DSS commands` info belongs to the `Manage DSS` how-to page. Added the content there.

File: docs/how-to/dss.rst [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+                            './kubeconfig'.
+        --all              Print the logs for all notebooks and MLflow.
+        --mlflow           Print the logs for the MLflow deployment.
+        --help             Show this message and exit.
+
+See also
+--------
+
+To learn how to manage your Jupyter Notebooks, check :ref:`manage_notebooks`. 
+If you are interested in managing MLflow within your DSS environment, see :ref:`manage_MLflow`.
```
Comment (kimwnasptd) [inline docs/how-to/dss.rst:181]: Should we have this as a list instead? My personal preference is that it's a bit more readable, but don't have a hard opinion.

Was looking at the `Next Steps` from MicroK8s as a nice example IMO
https://microk8s.io/docs/getting-started

File: docs/how-to/jupyter-notebook.rst [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+
+    Before creating a notebook, you need to select an image that includes the packages and toolkits you need.  
+    To see a list of recommended images and their aliases, do:
+
+    .. code-block:: bash
+
+        dss create --help
+
+    The output includes a list of recommended images and their aliases.
+    For example, this guide uses the image `kubeflownotebookswg/jupyter-scipy:v1.8.0`
```
Comment (kimwnasptd) [inline docs/how-to/jupyter-notebook.rst:26]: Same here for indentation, should we have a uniform story for lists that have a lot of text and code blocks?

File: docs/how-to/jupyter-notebook.rst [hunk 3]
```diff
@@ -0,0 +1,380 @@
...
+    [INFO] [I 2024-04-30 13:44:21.012 ServerApp] Package jupyter_server_ydoc took 0.0105s to import
+    [INFO] [I 2024-04-30 13:44:21.022 ServerApp] Package jupyterlab_git took 0.0104s to import
+    [INFO] [I 2024-04-30 13:44:21.022 ServerApp] Package nbclassic took 0.0000s to import
+
+.. _notebook-mlflow:
+
+Connect from notebook to MLflow
+-------------------------------
+
+You can integrate `MLflow <Charmed MLflow_>`_ with your Jupyter Notebook for tracking experiments using DSS.
```
[the remark this answers is not in the document]
Comment (kimwnasptd) [reply docs/how-to/jupyter-notebook.rst:278]: Agree as well, and then from the Charmed MLflow ones we could link to the upstream MLflow ones for useful information

**Review by kimwnasptd — COMMENTED** [review]
Did a first pass, nice work!

File: docs/how-to/dss.rst [hunk 4]
```diff
@@ -0,0 +1,181 @@
...
+        Commands:
+        create      Create a Jupyter notebook in DSS and connect it to MLflow.
+        initialize  Initialize DSS on the given Kubernetes cluster.
+        list        Lists all created notebooks in the DSS environment.
+        logs        Prints the logs for the specified notebook or DSS component.
+        purge       Removes all notebooks and DSS components.
+        remove      Remove a Jupter Notebook in DSS with the name NAME.
+        start       Starts a stopped notebook in the DSS environment.
+        status      Checks the status of key components within the DSS...
+        stop        Stops a running notebook in the DSS environment.
```
[the remark this answers is not in the document]
Comment (afgambin) [reply docs/how-to/dss.rst:144]: Replying to the indentation comments, yes, I think it looks nicer without it. I have removed it.

**Comment by afgambin** [comment]
@kimwnasptd ready for another pass/approval!

**Comment by kimwnasptd** [comment]
@afgambin looks good! Let's rebase now that https://github.com/canonical/data-science-stack/pull/131 was merged and we are good to go

**Review by kimwnasptd — APPROVED** [review]

**Pull request merged**